modules/friendlytag.js

```diff
@@ -240,6 +240,26 @@ Twinkle.tag.callback = function friendlytagCallback() {
 		}
 	});
 
+	// Hacky handling of form submission when quickfilter hides required fields that weren't filled in:
+	// Remove the form submit handler
+	/** @param {HTMLFormElement} form */
+	var submitForm = function(form) {
+		if (form.checkValidity()) {
+			Twinkle.tag.callback.evaluate({target: form});
+		} else {
+			$(form.quickfilter).val('');
+			// XXX: $(form.quickfilter).keyup() doesn't seem to work, so manually doing what the
```

Comment:
Not sure what you mean here?

Reply:
The main quickfilter function is a keyup handler on the input field. So I'd have expected `$(form.quickfilter).val('').keyup()` or `$(form.quickfilter).val('').trigger('keyup')` to automatically [make all checkboxes and headers show up](https://github.com/azatoth/twinkle/blob/3813f112ff2d7475dcc2d9b43239e4b6761043e6/modules/friendlytag.js#L82), but it doesn't work for some reason.

Reply:
Ah okay I get you now.  I know `.trigger` is imperfect, looks like `.keyup` is not calling the event at all.  IIRC events added via `addEventListener` are difficult to find, so that could be the root of it.